test/MessageTraitTest.php

```diff
@@ -66,6 +65,27 @@ public function testWithProtocolVersionRaisesExceptionForInvalidVersion($version
         $request->withProtocolVersion($version);
     }
 
+    public function validProtocolVersionProvider()
+    {
+        return [
+            '1.0' => [ '1.0' ],
+            '1.1' => [ '1.1' ],
+            '2'   => [ '2' ],
+            '2.0' => [ '2.0' ],
+        ];
+    }
+
+    /**
+     * @dataProvider validProtocolVersionProvider
+     */
+    public function testWithProtocolVersionDoesntRaiseExceptionForValidVersion($version)
+    {
+        $request = new Request();
+
+        $request->withProtocolVersion($version);
+        $this->addToAssertionCount(1);
```

Comment:
We should check the request protocol version at this point (instead of incrementing assertion counter)